Lib/test/libregrtest/main.py

```diff
@@ -461,6 +463,7 @@ def display_summary(self) -> None:
         print()
         print("Total duration: %s" % format_duration(duration))
 
+        assert self.first_runtests, "self.first_runtests is None"
```

Comment:
This assertion message doesn't tell the user about the programming error that meant that we hit the assert. If I understand correctly, the implicit invariant here is that you should never call the `display_summary` method before calling the `_run_tests` method (which sets `self.first_runtests` to a non-`None` object). So maybe we could say that in the assertion message?

```suggestion
        assert self.first_runtests, "Should never call `display_summary()` before calling `_run_test()`"
```